### DISCUSSION TITLE
Unable to convert after 8.2 upgrade

### DISCUSSION THREAD

**Discussion by renatobarcellos** [aclap-dev/video-downloadhelper#204] [Help and Q&A] [answered]
### Your question

After the latest update (8.2.1.1) I'm unable to convert mp4 videos. I used to convert videos after downloading them using Output configuration: "Re-encoded MP4 (h264/aac)". After selecting the local file and, picking the Output configuration, and clicking the green "Convert" button, another screen would come up so I could select where to save the converted file, and I could convert with no issues. Ever since I upgraded to 8.2.1.1, after I click "Convert", nothing happens: no option comes up to pick the file location, and the conversion doesn't start either. I'm not sure what's missing, I was able to convert like that almost daily. Thanks in advance.

I have

### Addon & CoApp version and details

Video DownloadHelper
Version 8.2.1.1 (stable)
Browser locale: en-US
Production build
Built on 2024-02-19
Build options: browser=chrome, linuxlic=false, noyt=true, target=google

Platform Mac arm64
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36

Found companion app: VdhCoApp 2.0.19
Companion app binary: /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp

License verified

**Comment by jcfrog** [comment]
yep, same here with 8.2.1.3a1 (beta)/2.0.19 Firefox MacOS 14.2.1
Tried to check the new "Use native filepicker" option, same thing, it just quits.

**Comment by GrampaWildWilly** [comment]
I'm on Windows 11 Pro 64-bit version 23H2 build 22631.3155, Firefox 123.0 64-bit, licensed VDH 8.2.1.3a1 beta, CoApp 2.0.19.  And I don't work for VDH.  I'm just another user.

I tried to convert a file just to see what I could see.  Conversion is something I pretty much never do, except when I'm investigating a problem for another user.  I started the Convert & the first problem I noticed is that it uses the old VDH file selection dialog instead of the native operating system file selection dialog, despite the fact that I have checked the box for VDH to use the system's native file selection dialog.  Anyway, after I selected the same target type for the conversion, the window just closed & nothing was going on.

So you've got 2 confirmations from 2 different operating systems.  There's something not right here.

**Comment by silvafabio** [comment, marked as answer]
Hi

The problem happen because the destination file has the same name and extension os the source file. I don't know how it was done before, but with new 8.2.1.1 version it starts to happen.

To replicate the problem, click in  "Download & Convert" and select "Re-encode MP4 (h264/aac)". After the download, when the conversion starts a red circle with error will be presented: 

> Convert2 error:
> Output /home/user/dwhelper/video.mp4 same as Input #0 - exiting
> Error opening output file /home/user/dwhelper/video.mp4.
> Error opening output files: Invalid argument

As workaround you can:
1. "Download" the video
2. Goto to the download folder and change the file extension from mp4 to mp4x or something else
3. Back in the "Video DownloadHelper" click in "Convert Local Files"
4. Select the renamed file
5. Select "Re-encode MP4 (h264/aac)"
6. Click in the Convert button
7. After the conversion you can delete the renamed (mp4x) file.

Looking forward to the fix ;-)

**Reply by Mattmjs** [reply]
I wonder why you need to download and convert, isn't the file already good enough as an mp4 and the codecs most of the
time are h264 + aac

**Reply by silvafabio** [reply]
In my case, when the download finish, the video is freezing and audio dyssynchronous with video.

Doing that process fix the video.

Look at my other topic https://github.com/aclap-dev/video-downloadhelper/discussions/64#discussioncomment-8428431

**Reply by renatobarcellos** [reply]
@silvafabio, you are right, the problem happens because the source and target files have the same name. Before, there would be an intermediate step where you could browse to save your destination file in a different folder and/or change the file name. This pop up/screen no longer shows, causing the error. For me, usually no error shows, it simply quits.
Your workaround idea is very useful and it does work. I'll mark your response as "the answer" because it provides a way of converting the files, which is my main goal here. I don't like doing workarounds when the feature was there in previous versions, but nonetheless, at least, it allows me to convert.

I was also playing around and was able to convert by first converting to .MOV, then back to .MP4 using "Re-encode MP4 (h264/aac)" option, but your solution saves me one extra step. Thanks!

**Reply by renatobarcellos** [reply]
@Mattmjs, I'm downloading classes that I take online every weekday that are 3-4 hours long. I have 2 reasons to convert the files:
1. For some reason, the .MP4 that I download for these classes only plays correctly using VLC Player. If I open it on my Mac (Final Cut Pro, Preview, etc), it skips, audio disappears, and once I convert them, these glitches are fixed (I don't know the reason for the glitches, or why they're fixed once converted)
2. The original size of each file is 5-6Gb. When I convert they drop to around 1.2-1.5Gb, saving a lot of disk space

**Reply by silvafabio** [reply]
Hi @renatobarcellos

In my case (chrome last version working) I used to click in "Download and Convert", and after that a popup with "Download & Convert" name show to select the desired conversion output format, so I select "Re-encoded MP4 (h264/aac)" click in Save, and after that i don't know how, the file is convert overriding the original file without any other confirmation (overriding) or popups.

**Comment by paulrouget** [comment]
@renatobarcellos @silvafabio 

Do you guys mind trying version 8.2.1.5a1 for Firefox: http://www.downloadhelper.net/data/vdh-mozilla-beta-latest.xpi - I think it should fix the issue.

If you use Chrome, I'll update the Chrome version too, but this will take a couple of days.

**Reply by jcfrog** [reply]
It seems to work better.
Still fails sometimes, for instance if the file picker tells me "file already exists", if I confirm I want to replace it, it doesn't, just quits

**Reply by GrampaWildWilly** [reply]
When I launch a conversion, the file selection dialog is the old-fashioned version.  After I select an output format & click the green button, the second file selection dialog is the native operating system file picker.  Way weird.  In any case, it works.

The status menu does not show a percentage complete or estimated time to completion or any of the usual things you'd expect on a status menu.  It just says Converting... - local convert.  It shows a progress bar but none of the other things.

I think with the Download and Convert function, you need to put in a check that makes the output file name different from the input file name.  The user doesn't get a chance between the time the download completes & the automatic convert step to supply a target file name for the converted file.  You should do something like stick the word Converted in front of the file name of the downloaded file.  But for a Convert Local File function, it's on the user's head to make sure the target of the conversion isn't also the source.  In fact, if the user accidentally does that, you shouldn't let the operation proceed.  Force the user to specify a different file for the output.

As I was testing this, I kept choosing files that were too ridiculously big.  So I cancelled a couple of conversions.  When I did the cancel, VDH gave me its usual completion notification.  But when I finally picked a file that was reasonably small & let it run to completion, I did not get the completion notification.  My system fans stopped blowing up a hurricane & the blue number on the VDH icon disappeared, but I didn't get the usual notification.

I tried the test Jérôme mentioned.  I converted the same file a second time to the same target without having first deleted the target file.  I got the prompt for duplicate file name, chose Yes, & it ran again, same as the first time.  But then, I'm using the native file picker.  Maybe the problem occurs when you use the old-fashioned file picker.  Which, like I said above, shouldn't be happening for me since I've got my VDH setting checked for using the native file picker.

**Reply by silvafabio** [reply]
I use Chrome.

Let me know when a version for test is available.

Thanks

**Reply by tomdfrog** [reply]
Hi.  The Beta did not fix it on my Mac.  MacOS Ventura on an Intel Mac running Firefox 123.  Installed it.  Quit and Relaunched Firefox.  Same problem.

To recap, click on the circle of arrows under VDH in my Toolbar.  Select the files I want to Convert.  Specify re-encode and click Convert.  When Convert is clicked, the window disappears.  It used to bring up a new Window for me to select the destination folder.  I have even tried disabling the Pop-up blocking in Firefox but that did not help.  

And to those who asked...I use this function to fix files previously downloaded which do not play well in the MacOS QuickView.  Yes, they play fine in VLC, but much of my viewing and sorting is in QuickView.

**Reply by renatobarcellos** [reply]
@paulrouget , I use Chrome and do not have a conversion license for Firefox. Please let us know once a fix is available and I'll gladly test it.

**Comment by tomdfrog** [comment]
Thanks.  Yes, adding a character to the .mp4 extension, such as .mp4**x**, before selecting the files to convert...solves the problem.  Hopefully either the choose destination dialog can be put back in OR by default, VDH simply puts Converted files in a new folder called Converted within the Source folder.  Both of which nix the clash of multiple files with the same name.

**Comment by paulrouget** [comment]
> To recap, click on the circle of arrows under VDH in my Toolbar. Select the files I want to Convert. Specify re-encode and click Convert. When Convert is clicked, the window disappears. It used to bring up a new Window for me to select the destination folder. I have even tried disabling the Pop-up blocking in Firefox but that did not help.

@tomdfrog You're seeing this behavior with 8.2.1.5? Because here it seems to work with 8.2.1.5.

**Comment by tomdfrog** [comment]
Yesterday...no.  Today...yes  Weird.  I even quit and relaunched Firefox yesterday as well.

Only difference,... after it failed yesterday, I had some trouble with a clip on YT, which was odd, so I uninstalled VDHbeta...then reinstalled the latest stable.  That prompted an update to the companion app as VDH said my companion app was out of date.  Now the Convert process brings back up the Save to dialog after clicking Convert.

FYI, the trouble with the YT video was not the Beta, but something with YT.  It will let me download that clip in HD but not 4K with both the Stable Release and the Beta.  Other 4K stuff on YT downloads just fine, so it it something screwy with that clip.

**Comment by tomdfrog** [comment]
Actually, it is irregular.  Earlier, it gave me the window to choose a destination, now it simply began converting the files after I clicked Convert...just with -converted added to the file name.

Also, it is not wanting to acknowledge my Prefs change to convert 4 not 1 file at  a time.  I have 10 cores and am stepping away from my computer.  I want it use all the power of my machine.

**Reply by GrampaWildWilly** [reply]
Can you provide some URLs?  But only if they can be accessed without registering for some sites.  We need URLs that don't require a logon.  Paul released a VDH 8.2.2.0a1 beta last night.  The CoApp is still 2.0.19.  Maybe these latest releases will do the trick.  In any case, I'd like to test the things that are failing for you.

**Reply by GrampaWildWilly** [reply]
Wait.  That's a dumb request.  Sorry.  You're talking about files local to your computer, of course.  I just tried converting a single file & it gave me 2 file selection dialogs.  The first one is incorrect.  It is the old-fashioned file picker.  It should be my native operating system file picker because I have enabled that new VDH setting.  The second file selection dialog is correctly the operating system file picker.  So if you do one at a time, you can convert from one file to a different one, even in a different directory.  If you select a different directory as the target, you can keep the same file name for the converted file.  But if you select multiple files to convert, I see 2 problems.  First, as you pointed out, they get done one at a time.  Second, also as you pointed out, the second file selection dialog does not appear.  However, the converted files all appear with a pattern like:

Original file name-converted.mp4

The file name gets modified with -converted & the file type is whatever you specified in your conversion rule.  The multiple converted files all appear in the same directory with the originals, which are all correctly preserved.  You really ought to get a chance to specify an output directory.  I could accept that all the converted files will go in the same directory, but it would be nice to be able to specify a different target directory for the output, same as you can when you do one at a time.  If the user specifies a different target directory, then you don't need to change the file names.  If the user specifies the same directory for both source & target, then the -converted modification to the file name would be the thing to do.  I think it would be beyond the point of diminishing returns to allow the user to select a different target directory for each individual conversion.  Just go to a single target directory & leave it to the user to scatter the converted files about to multiple target directories himself after the fact.

Further to the conversion making the first file selection dialog be of the wrong style, you should separate that into 2 steps.  The first step should be some kind of custom dialog that allows the user to select the target type for the conversion.  After the user sets the target type for the conversion, you should open the operating system file selection dialog, since that's what's in my settings.  That dialog is perfectly adapted to multiple file selection.  That is accomplished via Ctrl+Spacebar, then Ctrl+Down/UpArrow|PageUp/Down, then another Ctrl+Spacebar, as many times as you like.  You can even do Ctrl+Shift instead of just Ctrl to select multiple consecutive files from the directory.  Or Ctrl+Click or Ctrl+Shift+Click for mouse-dependent people.  So you can handle both single-file & multi-file selection of the source files for converting using the operating system file selection dialog.  After you make your selections, you can scroll the window while holding down Ctrl to review your choices & make any changes you might decide to make.  Then after that dialog gets committed by the user hitting Enter or clicking a Save button, you should pop up another operating system file selection dialog for setting the target directory of the conversion.  We need 2 operating system file selection dialogs.  What we have now is the home-grown file selection dialog followed by the operating system file selection dialog, which is wrong.

**Comment by Catagina** [comment]
Hello.
I have the same problem: after the latest update (8.2.1.1) converter don't work.
Will the future update fix the bug?
Thanks

**Reply by GrampaWildWilly** [reply]
You can do conversions, although with some issues as I detailed above, in the latest beta for Firefox.  That means you will be able to do conversions in some release of VDH on the other browsers.  It may already be available in a beta for the other browsers.  Since I don't work for VDH, & since I don't use anything other than Firefox, I don't know.  You can try it & see, I suppose.  Or just wait for an answer.  It is the weekend, so @paulrouget might not be back in here until Monday.

**Reply by Catagina** [reply]
Thanks for your reply.

I'm using Chrome. Before payment everything worked successfully. Then they asked me to pay. I was sad, but I paid for it and received a non-working converter that gives an error.
